### PR TITLE
update addonconfig when search operator is updated

### PR DESCRIPTION
**Pull request by SherinV**
### Related Issue

https://issues.redhat.com/browse/ACM-8830

### Description of changes
- Added ...

### CODE REVIEW

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

File: controllers/create_addondeploymentconfig.go [hunk 1]
```diff
@@ -34,10 +33,20 @@ func (r *SearchReconciler) createAddOnDeploymentConfig(ctx context.Context,
 		log.Error(err, "Error getting AddOnDeploymentConfig")
 		return &reconcile.Result{}, err
 	}
+	if !AddonDeploymentConfigEquals(found, adc) {
+		//get the current resource version to update
+		adc.ResourceVersion = found.ResourceVersion
```
Comment (SherinV) [inline controllers/create_addondeploymentconfig.go:38]: Copied over the resource version from existing AddOnDeploymentConfig as i was getting an error on update - Could not update AddOnDeploymentConfig - missing resourceversion. Please review. Thanks, @jlpadilla

**Pull request merged**